src/main/app/claims/responseModelConverter.ts

```diff
@@ -10,86 +13,205 @@ import { Company } from 'app/claims/models/details/yours/company'
+      }
+      break
+    case PartyType.SOLE_TRADER_OR_SELF_EMPLOYED.value:
+      party = new SoleTrader()
+      if ((defendant.partyDetails as SoleTraderDetails).businessName) {
+        (party as SoleTrader).businessName = (defendant.partyDetails as SoleTraderDetails).businessName
+      }
+      break
+  }
+  party.address = new Address().deserialize(defendant.partyDetails.address)
+  if (defendant.partyDetails.hasCorrespondenceAddress) {
+    party.correspondenceAddress = new Address().deserialize(defendant.partyDetails.correspondenceAddress)
+  }
+  party.name = defendant.partyDetails.name
+  if (defendant.email) {
+    party.email = defendant.email.address
+  }
+  if (defendant.mobilePhone) {
+    party.mobilePhone = defendant.mobilePhone.number
+  }
+  return party
+}
+
+function convertDefenceType (draft: RejectAllOfClaim): DefenceType {
+  switch (draft.option) {
+    case RejectAllOfClaimOption.ALREADY_PAID:
+      return DefenceType.ALREADY_PAID
+    case RejectAllOfClaimOption.DISPUTE:
+      return DefenceType.DISPUTE
+  }
+
+  throw new Error(`Unsupported defence type: ${draft.option}`)
+}
+
+function convertPartAdmissionType (draft: RejectPartOfClaim): PartAdmissionType {
+  switch (draft.option) {
+    case RejectPartOfClaimOption.AMOUNT_TOO_HIGH:
+      return PartAdmissionType.AMOUNT_TOO_HIGH
+    case RejectPartOfClaimOption.PAID_WHAT_BELIEVED_WAS_OWED:
+      return PartAdmissionType.PAID_WHAT_BELIEVED_WAS_OWED
+  }
+}
+
+function convertHowMuchOwed (draft: DraftHowMuchOwed): HowMuchOwed {
+  return {
+    amount: draft.amount,
+    explanation: draft.text
+  }
+}
+
+function convertTimeline (draft: DraftTimeline): Timeline {
+  return draft.getPopulatedRowsOnly()
+  .map(row => ({
+    date: row.date,
+    description: row.description
+  }))
+}
+
+function convertEvidence (draft: DraftEvidence): Evidence {
+  return draft.getPopulatedRowsOnly()
+  .map(row => ({
+    type: mapEvidenceType(row.type),
+    description: row.description
+  }))
+
+  function mapEvidenceType (draft: DraftEvidenceType): EvidenceType {
+    switch (draft.value) {
+      case DraftEvidenceType.CONTRACTS_AND_AGREEMENTS.value:
+        return EvidenceType.CONTRACTS_AND_AGREEMENTS
+      case DraftEvidenceType.EXPERT_WITNESS.value:
+        return EvidenceType.EXPERT_WITNESS
+      case DraftEvidenceType.CORRESPONDENCE.value:
+        return EvidenceType.CORRESPONDENCE
+      case DraftEvidenceType.PHOTO.value:
+        return EvidenceType.PHOTO
+      case DraftEvidenceType.RECEIPTS.value:
+        return EvidenceType.RECEIPTS
+      case DraftEvidenceType.STATEMENT_OF_ACCOUNT.value:
+        return EvidenceType.STATEMENT_OF_ACCOUNT
+      case DraftEvidenceType.OTHER.value:
+        return EvidenceType.OTHER
     }
-    if (defendant.mobilePhone) {
-      party.mobilePhone = defendant.mobilePhone.number
+  }
+}
+
+function convertPaymentPlan (draft: DefendantPaymentPlan): PaymentPlan {
+  if (validator.validateSync(draft).length > 0) {
```

Comment:
I don't like the fact validator is used to make conversion. Converter should take what is given and validate it without any additional check. Alternatively we can build contract in such a way that converter validates object on the `fromDraft` method and throws an exception when invalid.